src/utils/Theme/helpers/mergeDeep.ts

```diff
@@ -1,36 +1,16 @@
-import { cloneDeep } from './cloneDeep';
-import { isObject } from './isObject';
+import mergeWith from 'lodash.mergewith';
 
 /**
  * Merge and deep copy the values of all the enumerable own properties of target object from source object to a new object
  * @param target The target object to get properties from.
  * @param source The source object from which to copy properties.
+ * @param mergeFunction
  * @return A new merged and deep copied object.
  */
 export function mergeDeep<T extends object, S extends object>(
   target: T,
-  source: S
+  source: S,
+  mergeFunction?: (objValue: any, srcValue: any, key: string) => string
 ): T & S {
-  if (isObject(source) && Object.keys(source).length === 0) {
-    return cloneDeep({ ...target, ...source });
-  }
-
-  const output = { ...target, ...source };
-
-  if (isObject(source) && isObject(target)) {
-    for (const key in source) {
-      if (isObject(source[key]) && key in target && isObject(target[key])) {
-        (output as Record<string, unknown>)[key] = mergeDeep(
-          target[key] as object,
-          source[key] as object
-        );
-      } else {
-        (output as Record<string, unknown>)[key] = isObject(source[key])
-          ? cloneDeep(source[key])
-          : source[key];
-      }
-    }
-  }
-
-  return output;
+  return mergeWith(target, source, mergeFunction);
```

Comment:
This feels dangerous to change. I'm sure we wrote this custom for a reason so I'm wondering what that was in the first place. We also need to make sure we have good tests for this before merging.l

Reply:
Added several tests.